fbssdc/bpy.py

```diff
@@ -42,13 +43,23 @@ def encode_dir(dict_file, binjs_encode, in_path, out_path):
       # 2. Extract AST
       print ('Preprocessing {}'.format(source_path))
       process = subprocess.run([binjs_encode, '--quiet', '--show-ast', '--in', source_path, '--out', ignored_out_directory.name], capture_output=True)
-      proggy = json.loads(process.stdout.decode('utf-8'))
+      try:
+        proggy = json.loads(process.stdout.decode('utf-8'))
 
-      # 3. Encode
-      dest_path = os.path.join(dest_root,source[:-3] + '.binjs')
-      print ('Encoding {source_path} => {dest_path}'.format(source_path=source_path, dest_path=dest_path))
-      dest_file = open(dest_path, 'wb')
-      format.write(types, string_dict, ty_script, proggy, dest_file)
+        # 3. Encode
+        dest_path = os.path.join(dest_root,source[:-3] + '.binjs')
+        print ('Encoding {source_path} => {dest_path}'.format(source_path=source_path, dest_path=dest_path))
+        dest_file = open(dest_path, 'wb')
+        format.write(types, string_dict, ty_script, proggy, dest_file)
+
+        # 4. Copy source file
+        if copy_source:
+          shutil.copy(source_path, dest_root)
+      except:
+        if skip_errors:
+          print ('...does not parse')
+        else:
+          pass
```

Comment:
So skip_errors=True prints "does not parse" but skip_errors=False just causes it to fail silently and continue with the for-loop? Don't you want the operation to stop if skip_errors=False?

Reply:
Oops, you're right.

I meant to rethrow instead of `pass`.